tardis/apps/publication_workflow/static/js/lib/angular-material-data-table-0.10.10/dist/md-data-table.js

```diff
@@ -0,0 +1,1486 @@
+
+  // empty controller to bind scope properties to
+  function Controller() {
+
+  }
+
+  function postLink(scope, element, attrs, ctrls) {
+    var self = ctrls.shift();
+    var tableCtrl = ctrls.shift();
+    var getId = $parse(attrs.mdSelectId);
+
+    self.id = getId(self.model);
+
+    if(tableCtrl.$$rowSelect && self.id) {
+      if(tableCtrl.$$hash.has(self.id)) {
+        var index = tableCtrl.selected.indexOf(tableCtrl.$$hash.get(self.id));
+
+        // if the item is no longer selected remove it
+        if(index === -1) {
+          tableCtrl.$$hash.purge(self.id);
+        }
+
+        // if the item is not a reference to the current model update the reference
+        else if(!tableCtrl.$$hash.equals(self.id, self.model)) {
+          tableCtrl.$$hash.update(self.id, self.model);
+          tableCtrl.selected.splice(index, 1, self.model);
+        }
+
+      } else {
+
+        // check if the item has been selected
+        tableCtrl.selected.some(function (item, index) {
+          if(getId(item) === self.id) {
+            tableCtrl.$$hash.update(self.id, self.model);
+            tableCtrl.selected.splice(index, 1, self.model);
+
+            return true;
+          }
+        });
+      }
+    }
+
+    self.isSelected = function () {
+      if(!tableCtrl.$$rowSelect) {
+        return false;
+      }
+
+      if(self.id) {
+        return tableCtrl.$$hash.has(self.id);
+      }
+
+      return tableCtrl.selected.indexOf(self.model) !== -1;
+    };
+
+    self.select = function () {
+      if(self.disabled) {
+        return;
+      }
+
+      if(tableCtrl.enableMultiSelect()) {
+        tableCtrl.selected.push(self.model);
+      } else {
+        tableCtrl.selected.splice(0, tableCtrl.selected.length, self.model);
+      }
+
+      if(angular.isFunction(self.onSelect)) {
+        self.onSelect(self.model);
+      }
+    };
+
+    self.deselect = function () {
+      if(self.disabled) {
+        return;
+      }
+
+      tableCtrl.selected.splice(tableCtrl.selected.indexOf(self.model), 1);
+
+      if(angular.isFunction(self.onDeselect)) {
+        self.onDeselect(self.model);
+      }
+    };
+
+    self.toggle = function (event) {
+      if(event && event.stopPropagation) {
+        event.stopPropagation();
+      }
+
+      return self.isSelected() ? self.deselect() : self.select();
+    };
+
+    function autoSelect() {
+      return attrs.mdAutoSelect === '' || self.autoSelect;
+    }
+
+    function createCheckbox() {
+      var checkbox = angular.element('<md-checkbox>').attr({
+        'aria-label': 'Select Row',
+        'ng-click': '$mdSelect.toggle($event)',
+        'ng-checked': '$mdSelect.isSelected()',
+        'ng-disabled': '$mdSelect.disabled'
+      });
+
+      return angular.element('<td class="md-cell md-checkbox-cell">').append($compile(checkbox)(scope));
+    }
+
+    function disableSelection() {
+      Array.prototype.some.call(element.children(), function (child) {
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 16 spaces but found 6.](https://app.codacy.com/gh/mytardis/mytardis/pullRequest?prid=7933947)
```suggestion
                Array.prototype.some.call(element.children(), function (child) {
```